### ISSUE TITLE
Bug in from_row_iter_dense()

### ISSUE THREAD

**Issue by yongqli** [oxfordcontrol/osqp.rs#15]
In the line `data[c * ncols + r] = iter.next().expect("not enough elements in iterator");`, it should be `data[c * nrows + r]` instead.

You can check it by running:

```
let (nrows, ncols) = (3, 2);
let mut v = vec![(0, 0); nrows * ncols];
for r in 0..nrows {
    for c in 0..ncols {
        v[c * nrows + r] = (r, c);
    }
}
let mut v = v.into_iter();
for c in 0..ncols {
    for r in 0..nrows {
        assert_eq!(v.next().unwrap(), (r, c));
    }
}
```

in the [Rust Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=c116b23a3c0321600d3633eb7165e1f9).

**Comment by w4rum** [comment]
Hey, sorry to necro this. Is there any chance we could get this pushed downstream? I ran into exactly this issue and was quite confused even after reading the documentation. Just using this Git repository in my `Cargo.toml` fixes this for the time being. If you could include this fix in a proper release, I'd really appreciate it.

**Comment by ebarnard** [comment]
I was hoping for an upstream OSQP release, but it seems one is not forthcoming.

This fix has been included in v0.6.1.

**Comment by w4rum** [comment]
Thanks a lot! This really helps.
